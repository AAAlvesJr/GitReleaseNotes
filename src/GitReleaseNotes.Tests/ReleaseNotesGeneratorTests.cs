using System;
using ApprovalTests;
using GitReleaseNotes.Git;
using GitReleaseNotes.IssueTrackers;
using LibGit2Sharp;
using Xunit;

namespace GitReleaseNotes.Tests
{
    public class ReleaseNotesGeneratorTests
    {
        [Fact]
        public void AllTagsWithNoCommitsOrIssuesAfterLastRelease()
        {
            IRepository repo;
            IIssueTracker issueTracker;
            new TestDataCreator(new DateTimeOffset(2012, 1, 1, 0, 0, 0, new TimeSpan()))
                .CreateRelease("0.1.0", "Issue1", "Issue2")
                .CreateRelease("0.2.0", "Issue3")
                .Build(out repo, out issueTracker);

            var tagToStartFrom = GitRepositoryInfoFinder.GetFirstCommit(repo);
            var currentReleaseInfo = GitRepositoryInfoFinder.GetCurrentReleaseInfo(repo); 

            var releaseNotes = ReleaseNotesGenerator.GenerateReleaseNotes(
                repo, issueTracker, new SemanticReleaseNotes(), new string[0],
                tagToStartFrom, currentReleaseInfo);

            Approvals.Verify(releaseNotes.ToString());
        }

        [Fact]
        public void AllTags()
        {
            IRepository repo;
            IIssueTracker issueTracker;
            new TestDataCreator(new DateTimeOffset(2012, 1, 1, 0, 0, 0, new TimeSpan()))
                .CreateRelease("0.1.0", "Issue1", "Issue2")
                .CreateRelease("0.2.0", "Issue3")
                .AddIssues("Issue4")
                .Build(out repo, out issueTracker);

            var tagToStartFrom = GitRepositoryInfoFinder.GetFirstCommit(repo);
            var currentReleaseInfo = GitRepositoryInfoFinder.GetCurrentReleaseInfo(repo); 

            var releaseNotes = ReleaseNotesGenerator.GenerateReleaseNotes(
                repo, issueTracker, new SemanticReleaseNotes(), new string[0],
                tagToStartFrom, currentReleaseInfo);

            Approvals.Verify(releaseNotes.ToString());
        }

        [Fact]
        public void AppendOnlyNewItems()
        {
            IRepository repo;
            IIssueTracker issueTracker;
            new TestDataCreator(new DateTimeOffset(2012, 1, 1, 0, 0, 0, new TimeSpan()))
                .CreateRelease("0.1.0", "Issue1", "Issue2")
                .CreateRelease("0.2.0", "Issue3")
                .AddIssues("Issue4")
                .Build(out repo, out issueTracker);

            var tagToStartFrom = GitRepositoryInfoFinder.GetFirstCommit(repo);
            var currentReleaseInfo = GitRepositoryInfoFinder.GetCurrentReleaseInfo(repo);

            var previousReleaseNotes = SemanticReleaseNotes.Parse(@"# vNext


Commits: ...


# 0.2.0 (05 January 2012)

 - [2] - Edited Issue3

Commits: F8C83FD88C...CD7A26452D


# 0.1.0 (03 January 2012)

 - [0] - Edited Issue1
 - [1] - Edited Issue2

Commits: 75B636A423...D2F38F9B38");

            var releaseNotes = ReleaseNotesGenerator.GenerateReleaseNotes(
                repo, issueTracker, previousReleaseNotes, new string[0],
                tagToStartFrom, currentReleaseInfo);

            Approvals.Verify(releaseNotes.ToString());
        }
    }
}