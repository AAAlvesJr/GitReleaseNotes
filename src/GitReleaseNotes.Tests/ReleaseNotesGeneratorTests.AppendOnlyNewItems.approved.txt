# vNext

 - [3] - Issue4

Commits: DE67510945...DE67510945


# 0.2.0 (05 January 2012)

 - [2] - Edited Issue3

Commits: F8C83FD88C...CD7A26452D


# 0.1.0 (03 January 2012)

 - [0] - Edited Issue1
 - [1] - Edited Issue2

Commits: 75B636A423...D2F38F9B38
